File: Snake_Qt/Director.cpp
#include "Director.h"

Director *Director::instance = nullptr; //初始化类的单例句柄

Director::Director() :window(nullptr), nowScene(nullptr)
{
}

Director::~Director()
{
}

Director * Director::getInstance()
{
	if (instance == nullptr)
	{
		instance = new Director();
	}
	return instance;
}

void Director::setWindow(Window * window)
{
	this->window = window;
}

Window * Director::getWindow()
{
	return window;
}

void Director::setNowScene(Scene * scene)
{
	nowScene = scene;
}

Scene * Director::getNowScene()
{
	return nowScene;
}

File: Snake_PyQt/Director.py
#导演类
class Director:
    instance = None  #引用单例

    #用new方法以生成唯一单例
    def __new__(cls):
        if not cls.instance:
            cls.instance = super().__new__(cls)
            cls.instance.window = None  #游戏主窗口引用实例
            cls.instance.nowScene = None  #当前受响应场景
        return cls.instance